flocker/control/_protocol.py

```diff
@@ -109,6 +109,11 @@ class NodeStateCommand(Command):
     Used by a convergence agent to update the control service about the
     status of a particular node.
     """
+    # FLOC-1513
+    #
+    # Add another argument, an optional `ListOf(Unicode)` representing
```

Comment:
The point is that `discover_local_state` returns an arbitrary list of `IClusterSTateUpdater` providers, and that's what you need to send over the wire. You shouldn't special case datasets.